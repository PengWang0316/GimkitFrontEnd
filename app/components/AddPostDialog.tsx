import React, { memo, useState, useCallback } from 'react';
import { connect } from 'react-redux';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import {
  Button, Dialog, DialogActions, DialogContent, DialogTitle, TextField,
} from '@material-ui/core';
import { useTheme, makeStyles } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';
import ReactQuill from 'react-quill';

import { PostType } from '../store/Posts/types';
import { addNewPost as addNewPostAction } from '../store/Posts/actions';

// Css for Quill editor
require('react-quill/dist/quill.snow.css');

interface Props {
  isOpen: boolean;
  handleClose: (event: React.MouseEvent) => void;
  addNewPost: (post: PostType) => void;
}

const useStyles = makeStyles({
  editor: {
    height: 300,
    paddingBottom: 40,
    width: '100%',
  },
  title: {
    width: '100%',
    marginBottom: 20,
  },
});

export const AddPostDialog = ({ handleClose, isOpen = false, addNewPost }: Props) => {
  const classes = useStyles({});
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));
  const [content, setContent] = useState('');
  const [title, setTitle] = useState('');
  const [isSaveAvailable, setIsSaveAvailable] = useState(false);

  // handle content change
  const handleContentChange = useCallback((value: string, delta, source, editor) => {
    setContent(value);
    setIsSaveAvailable(title.length !== 0 && editor.getText().length !== (0 || 1));
  }, [content, title]);

  // handle title change
  const handleTitleChange = useCallback((event: React.ChangeEvent) => {
    const { value } = event.target as HTMLInputElement;
    setTitle((event.target as HTMLInputElement).value);
    setIsSaveAvailable(content.length !== 0 && value.length !== 0);
  }, [content, title]);

  // handle click save button
  const handleSave = () => {
    const post: PostType = { title, content };
    addNewPost(post);
    setTitle('');
    setContent('');
    handleClose(null);
  };

  return (
    <Dialog
      fullScreen={fullScreen}
      open={isOpen}
      onClose={handleClose}
      aria-labelledby="addPostDialogTitle"
      maxWidth="md"
    >
      <DialogTitle id="addPostDialogTitle">{I18n.get('addPostTitle')}</DialogTitle>
      <DialogContent>
        <TextField
          label={I18n.get('title')}
          className={classes.title}
          margin="dense"
          value={title}
          onChange={handleTitleChange}
        />
        <ReactQuill value={content} onChange={handleContentChange} className={classes.editor} />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleSave} color="secondary" disabled={!isSaveAvailable}>
          {I18n.get('save')}
        </Button>
        <Button onClick={handleClose} color="primary">
          {I18n.get('cancel')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  addNewPost: (post: PostType) => dispatch(addNewPostAction(post)),
});
export default connect(null, mapDispatchToProps)(memo(AddPostDialog));
