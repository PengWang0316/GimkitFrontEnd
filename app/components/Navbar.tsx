/* eslint-disable react/destructuring-assignment */
import React, { Fragment } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Toolbar, AppBar, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { orange, green } from '@material-ui/core/colors';
// import { Auth } from 'aws-amplify';
import I18n from '@kevinwang0316/i18n';

import { HOME_PAGE_URL } from '../config';

/* istanbul ignore next */
const useStyles = makeStyles(theme => ({
  link: {
    color: theme.palette.primary.contrastText,
    textDecoration: 'none',
  },
  menuLink: {
    textDecoration: 'none',
  },
  flex1: {
    flex: 1,
  },
  appbar: {
    maxHeight: 55,
    position: 'sticky',
    top: 0,
    zIndex: theme.zIndex.drawer + 1,
  },
  avatar: {
    width: 26,
    height: 26,
    marginRight: 8,
    color: '#fff',
    backgroundColor: orange[800],
  },
  switchBase: {
    color: 'white',
    '&$checked': {
      color: green[400],
    },
    '&$checked + $track': {
      backgroundColor: green[400],
    },
  },
  checked: {},
  track: {},
}));

/** Navbar component */
export const Navbar = () => {
  const classes = useStyles({});
  return (
    <Fragment>
      <AppBar position="static" className={classes.appbar} data-testid="navbar">
        <Toolbar>
          <Link to={HOME_PAGE_URL} className={`${classes.link} ${classes.flex1}`} data-testid="titleLink">
            <Typography variant="h6" color="inherit">{I18n.get('appName')}</Typography>
          </Link>
        </Toolbar>
      </AppBar>
    </Fragment>
  );
};

export default withRouter(Navbar);
