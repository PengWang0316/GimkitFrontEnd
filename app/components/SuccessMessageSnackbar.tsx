import React from 'react';
import {
  Snackbar, SnackbarContent, Icon, IconButton,
} from '@material-ui/core';
import { Close as CloseIcon } from '@material-ui/icons';

interface Props {
  isOpen: boolean;
  handleClose: () => void;
  message: string;
}

export const SuccessMessageSnackbar = ({ isOpen, handleClose, message }: Props) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    open={isOpen}
    autoHideDuration={3000}
    onClose={handleClose}
  >
    <SnackbarContent
      message={(
        <span>
          <Icon />
          {message}
        </span>
      )}
      action={[
        <IconButton key="close" aria-label="Close" color="inherit" onClick={handleClose}>
          <CloseIcon />
        </IconButton>,
      ]}
    />
  </Snackbar>
);

export default SuccessMessageSnackbar;
