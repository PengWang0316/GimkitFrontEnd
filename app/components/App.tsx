import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import importedComponent from 'react-imported-component';
import CssBaseline from '@material-ui/core/CssBaseline';

import Navbar from './Navbar';
import { HOME_PAGE_URL } from '../config';
import LoadingAnimation from './SharedComponents/LoadingAnimation';

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#4A6572',
      main: '#344955',
      dark: '#232F34',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
const HomePage = importedComponent(() => import(/* webpackChunkName: "HomePageContainer" *//* webpackPrefetch: true */ './containers/HomePageContainer').catch(err => console.log(err)));

/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export const App = () => (
  <MuiThemeProvider theme={theme}>
    <Router>
      <div>
        <CssBaseline />
        <Navbar />
        <main>
          <Switch>
            <Route exact path={HOME_PAGE_URL} component={HomePage} />
            <Route render={() => <p>Not Fount!</p>} />
          </Switch>
        </main>
      </div>
    </Router>
  </MuiThemeProvider>
);
export default App;
