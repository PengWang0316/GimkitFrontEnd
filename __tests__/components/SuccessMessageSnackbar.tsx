import React from 'react';
import renderer from 'react-test-renderer';

import { SuccessMessageSnackbar } from '../../app/components/SuccessMessageSnackbar';

jest.mock('@material-ui/core/Snackbar', () => 'Snackbar');
jest.mock('@material-ui/core/SnackbarContent', () => 'SnackbarContent');
jest.mock('@material-ui/core/Icon', () => 'Icon');
jest.mock('@material-ui/core/IconButton', () => 'IconButton');
jest.mock('@material-ui/icons/Close', () => 'CloseIcon');


describe('SuccessMessageSnackbar', () => {
  test('snapshot', () => expect(renderer.create(<SuccessMessageSnackbar {...{ isOpen: true, handleClose: jest.fn(), message: 'message' }} />).toJSON()).toMatchSnapshot());
});
