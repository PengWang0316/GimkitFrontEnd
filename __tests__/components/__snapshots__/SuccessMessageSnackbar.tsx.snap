// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessMessageSnackbar snapshot 1`] = `
<Snackbar
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "bottom",
    }
  }
  autoHideDuration={3000}
  onClose={[MockFunction]}
  open={true}
>
  <SnackbarContent
    action={
      Array [
        <IconButton
          aria-label="Close"
          color="inherit"
          onClick={[MockFunction]}
        >
          <CloseIcon />
        </IconButton>,
      ]
    }
    className="makeStyles-success-1"
    message={
      <span>
        <Icon />
        message
      </span>
    }
  />
</Snackbar>
`;
